apps/guardian-ui/src/GuardianApi.ts

```diff
@@ -47,6 +47,37 @@ class BaseGuardianApi
 {
   private websocket: JsonRpcWebsocket | null = null;
   private connectPromise: Promise<JsonRpcWebsocket> | null = null;
+  private websocketUrl: string | undefined;
+
+  constructor() {
+    // Attempt to get the websocketUrl from the environment variable
+    const websocketUrlFromEnv = process.env.REACT_APP_FM_CONFIG_API;
+
+    // If the environment variable is not set, try to fetch from config.json
+    if (websocketUrlFromEnv) {
+      this.websocketUrl = websocketUrlFromEnv;
+    } else {
+      this.retrieveWebsocketUrlFromConfig();
+    }
+  }
+
+  // Asynchronous method to fetch the websocketUrl from config.json
+  async retrieveWebsocketUrlFromConfig() {
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', process.env.PUBLIC_URL + '/config.json', false); // Synchronous request
```

Comment:
Never use synchronous XMLHttpRequest, it will block all execution while the request is in flight. This will result in a terrible user experience if we want to give the user any understanding of what's going on while this is loading.

Opening a socket is already asynchronous by nature, this doesn't need to be made synchronous to maintain the